src/main/java/uk/gov/hmcts/ccd/data/casedetails/query/CaseDetailsQueryBuilderFactory.java

```diff
@@ -3,24 +3,39 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import uk.gov.hmcts.ccd.data.casedetails.CaseDetailsEntity;
+import uk.gov.hmcts.ccd.data.casedetails.search.MetaData;
 
 import javax.persistence.EntityManager;
+import java.util.List;
 
 @Component
 public class CaseDetailsQueryBuilderFactory {
 
     private final UserAuthorisationSecurity userAuthorisationSecurity;
+    private final List<CaseDetailsAuthorisationSecurity> caseDetailsAuthorisationSecurities;
 
     @Autowired
-    public CaseDetailsQueryBuilderFactory(UserAuthorisationSecurity userAuthorisationSecurity) {
+    public CaseDetailsQueryBuilderFactory(UserAuthorisationSecurity userAuthorisationSecurity,
+                                          List<CaseDetailsAuthorisationSecurity> caseDetailsAuthorisationSecurities) {
         this.userAuthorisationSecurity = userAuthorisationSecurity;
+        this.caseDetailsAuthorisationSecurities = caseDetailsAuthorisationSecurities;
+    }
+
+    public CaseDetailsQueryBuilder<CaseDetailsEntity> select(EntityManager em, MetaData metaData) {
+        return secure(select(em), metaData);
```

Comment:
select and count methods are implemented in a slightly different way, but do that same thing. 
The both apply all security checks to the builder. 
It would make things simpler if both were implemented using the same approach

Reply:
reorganised code, count and select are consistent now